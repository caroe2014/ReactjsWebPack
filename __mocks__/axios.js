
const mockAxios = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.create = () => mockAxios;
mockAxios.get = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.post = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.put = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.delete = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.interceptors = {
  response: {
    use: jest.fn()
  },
  request: {
    use: jest.fn()
  }
};

export default mockAxios;
