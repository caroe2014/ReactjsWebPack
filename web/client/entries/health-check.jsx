// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled from 'styled-components';
import ReactDOM from 'react-dom';
import config from 'app.config';
import axios from 'axios';
import { Flex, Text, Box, Border, Pre, Link } from 'rebass';
import auth from 'web/client/auth';

const authrorized = auth.checkAuth();

const logger = config.logger.child({ childName: 'health-check' });
const serviceContext = `${config.webPaths.serviceContext}${config.webPaths.serviceContext ? `/` : ''}`;

const HealthCheckLink = styled(Link)`
  cursor: pointer;
`;

export class HealthCheck extends Component {
  constructor (props) {
    super(props);
    this.state = {
      loading: true,
      error: false,
      frameContent: null
    };

    this.updateHeathCheckSummary = this.updateHeathCheckSummary.bind(this);
    this.updateManifest = this.updateManifest.bind(this);
  }

  componentDidMount () {
    this.updateHeathCheckSummary();
  }

  handleClick (fn) {
    this.setState({
      loading: false,
      error: false
    }, fn);
  }

  updateHeathCheckSummary () {
    axios.get(`${serviceContext}internal/healthchecks`)
      .then((results) => {
        this.setState({
          frameContent: JSON.stringify(results.data, null, 2),
          loading: false
        });
      })
      .catch(error => {
        logger.error(error);
        this.setState({ error: true });
      });
  }

  updateManifest () {
    axios.get(`${serviceContext}internal/manifest`)
      .then((results) => {
        this.setState({
          frameContent: JSON.stringify(results.data, null, 2),
          loading: false
        });
      })
      .catch(error => {
        logger.error(error);
        this.setState({ error: true });
      });
  }

  render () {

    const loading = this.state.loading && !this.state.error ? (<Text> Loading... </Text>) : null;
    const error = this.state.error ? (<Text color='red'> Error loading health check information </Text>) : null;

    const pre = (<Pre style={{ whiteSpace: 'pre-wrap' }} children={this.state.frameContent} />);

    const frame = !this.state.loading && !this.state.error ? pre : null;

    return (
      <Flex flexWrap='wrap' flexDirection={['column', 'row']} px={10} py={20} style={{ height: '100%' }}>
        <Box flex='0 1 auto' mr={20} pb={10}>
          <ul>
            <li>
              <HealthCheckLink
                className='health-check-link'
                onClick={() => this.handleClick(this.updateHeathCheckSummary)} children='Health Check Summary' />
            </li>
            <li>
              <HealthCheckLink
                className='health-check-link2'
                onClick={() => this.handleClick(this.updateManifest)}
                children='Manifest'
              />
            </li>
          </ul>
        </Box>
        <Flex flex='1 1'>
          <Border className='text-here' p={10} w={'100%'}>
            {error}
            {loading}
            {frame}
          </Border>
        </Flex>
      </Flex>);
  }
}
if (process.env.NODE_ENV !== 'test') {
  const renderHealthcheck = async () => {
    if (await authrorized) {
      ReactDOM.render(
        <HealthCheck />,
        document.getElementById('appContainer')
      );
    }
  };
  renderHealthcheck();
}
