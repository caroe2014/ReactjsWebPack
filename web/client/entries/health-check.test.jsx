// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import axios from 'axios';
import Adapter from 'enzyme-adapter-react-16';

import { HealthCheck } from './health-check';

configure({
  adapter: new Adapter()
});

describe('HealthCheck', () => {

  it('renders correct snapshot', () => {
    const tree = renderer.create(
      <HealthCheck />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<HealthCheck />).exists()).toBe(true);
  });

  it('should render without throwing an error', () => {
    let wrapper = shallow(<HealthCheck />);
    wrapper.setState({ error: true });
    expect(shallow(<HealthCheck />).exists()).toBe(true);
  });

  it('should call handleClick and updateHeathCheckSummary when clicked', () => {
    let wrapper = shallow(<HealthCheck />);
    const spy = jest.spyOn(wrapper.instance(), 'handleClick');
    wrapper.instance().forceUpdate();
    wrapper.find('.health-check-link').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call handleClick and updateHeathCheckSummary when clicked but error is thrown', () => {
    let wrapper = shallow(<HealthCheck />);
    axios.get.mockReturnValue(Promise.reject(new Error('error')));
    const spy = jest.spyOn(wrapper.instance(), 'handleClick');
    wrapper.instance().forceUpdate();
    wrapper.find('.health-check-link').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call handleClick and updateManifestmockResponse when clicked', () => {

    let mockResponse = {
      data: {
        someData: 'testData'
      }
    };
    axios.get.mockReturnValue(Promise.resolve(mockResponse));

    let wrapper = shallow(<HealthCheck />);
    const spy = jest.spyOn(wrapper.instance(), 'handleClick');
    wrapper.instance().forceUpdate();
    wrapper.find('.health-check-link2').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call handleClick and updateManifestmockResponse when clicked but error is thrown', () => {
    let wrapper = shallow(<HealthCheck />);
    axios.get.mockReturnValue(Promise.reject(new Error('error')));
    const spy = jest.spyOn(wrapper.instance(), 'handleClick');
    wrapper.instance().forceUpdate();
    wrapper.find('.health-check-link2').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

});
