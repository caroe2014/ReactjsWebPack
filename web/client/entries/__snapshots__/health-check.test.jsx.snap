// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthCheck renders correct snapshot 1`] = `
<div
  className="sc-bwzfXH egOajC sc-bdVaJa kjejsN"
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    className="sc-bdVaJa jplJEP"
  >
    <ul>
      <li>
        <a
          className="health-check-link sc-jqCOkK dLGIUz sc-htoDjs gLCdRU"
          onClick={[Function]}
        >
          Health Check Summary
        </a>
      </li>
      <li>
        <a
          className="health-check-link2 sc-jqCOkK dLGIUz sc-htoDjs gLCdRU"
          onClick={[Function]}
        >
          Manifest
        </a>
      </li>
    </ul>
  </div>
  <div
    className="sc-bwzfXH hKiLMS sc-bdVaJa OqbGj"
  >
    <div
      className="text-here sc-jlyJG eGdVCw"
      w="100%"
    >
      <div
        className="sc-gqjmRU kKsieB"
      >
         Loading... 
      </div>
    </div>
  </div>
</div>
`;
